File: src/main/java/com/snackgame/server/auth/oauth/oidc/payload/KakaoIdTokenPayload.java
package com.snackgame.server.auth.oauth.oidc.payload;

import java.util.Map;

public class KakaoIdTokenPayload implements IdTokenPayload {

    private final Map<String, Object> payload;

    public KakaoIdTokenPayload(Map<String, Object> payload) {
        this.payload = payload;
    }

    @Override
    public String getProvider() {
        return "kakao";
    }

    @Override
    public String getId() {
        return payload.get("sub").toString();
    }

    @Override
    public String getEmail() {
        return payload.get("email").toString();
    }

    @Override
    public String getName() {
        return payload.get("name").toString();
    }

    @Override
    public String getPicture() {
        return (String)payload.getOrDefault("picture", null);
    }
}

File: src/main/java/com/snackgame/server/auth/oauth/oidc/payload/KakaoIdTokenPayload.kt
package com.snackgame.server.auth.oauth.oidc.payload

class KakaoIdTokenPayload(private val payload: Map<String, Any>) : IdTokenPayload {
    override val provider: String
        get() = "kakao"
    override val id: String
        get() = payload["sub"] as String
    override val email: String
        get() = payload["email"] as String
    override val name: String?
        get() = payload["name"] as String?
    override val picture: String?
        get() = payload["picture"] as String?
}
